e2e/test/FaultInjection.cs

```diff
@@ -62,5 +63,121 @@ public static Client.Message ComposeErrorInjectionProperties(string faultType, s
                 }
             };
         }
+
+        public static async Task ActivateFaultInjection(Client.TransportType transport, string faultType, string reason, int delayInSec, int durationInSec, DeviceClient deviceClient)
+        {
+            s_log.WriteLine($"{nameof(ActivateFaultInjection)}: Requesting fault injection type={faultType} reason={reason}, delay={delayInSec}s, duration={FaultInjection.DefaultDurationInSec}s");
+
+            uint oldTimeout = deviceClient.OperationTimeoutInMilliseconds;
+
+            try
+            {
+                // For MQTT FaultInjection will terminate the connection prior to a PUBACK 
+                // which leads to an infinite loop trying to resend the FaultInjection message.
+                if (transport == Client.TransportType.Mqtt ||
+                    transport == Client.TransportType.Mqtt_Tcp_Only ||
+                    transport == Client.TransportType.Mqtt_WebSocket_Only)
+                {
+                    deviceClient.OperationTimeoutInMilliseconds = 1000;
```

Comment:
Could this operation timeout be shorter? I imagine you will be sending and re-sending this error injection message a few times which may be hard to account for later.

Reply:
Good point: I think we'll need to revising timings in the future. Right now each test takes 10s...